### PR TITLE
Option to choose renderer (canvas or svg)

### PR DESCRIPTION
**Pull request by eterna2**
There is some bug in vega's `canvas` rendering of biezer curves using the `path` marks. You need `svg` rendering for the curves to show up correctly.

Added option to choose the renderer to use:
- either at creation `Vega(spec, renderer='svg')`
- or, at dispkat `vegaObj.display(renderer='svg')

Default for renderer will be `canvas`.

### CODE REVIEW

File: vega/base.py [hunk 1]
```diff
@@ -51,6 +53,8 @@ def _ipython_display_(self):
             metadata={'jupyter-vega': '#{0}'.format(id)}
         )
 
-    def display(self):
+    def display(self, renderer=None):
         """Render the visualization."""
+        if renderer:
```
Comment (domoritz) [inline vega/base.py]: Why is this necessary? Isn't `self.renderer` not defined the same as setting it to `None`?

File: vega/base.py [hunk 2]
```diff
@@ -51,6 +53,8 @@ def _ipython_display_(self):
...
+            self.renderer = renderer
```
Comment (rgbkrk) [inline vega/base.py]: This should only get set in the `__init__`, otherwise this is a side effect on the class when displayed.

Comment (eterna2) [inline vega/base.py]: I agree with that. 

But I was hoping for a way to quickly swap between the renderers without creating a new Vega object.

But ya, the side effect can be nasty.

File: vega/base.py (hunk 1 above)
Comment (eterna2) [inline vega/base.py]: Default renderer is `canvas`.
I don't want that to be overwritten if the optional arg is entered.

But on 2nd thoughts, I could just have set it to `canvas` instead of `none`.

But I think I will remove the arg entirely, until I find a way to pass down the options without side effects

**Comment by domoritz** [comment]
This was left open since there were some things left to do. Could you redo this for `vega3`?

**Comment by eterna2** [comment]
Sure. Sorry about the loose ends.

**Comment by pagpires** [comment]
Hi guys, thanks for your efforts to make ipyvega so nice. I'm just wondering about the status of renderer. It seems right now we still cannot choose a renderer by using like VegaLite(spec, renderer='svg') right? Or did I miss anything? Thanks a lot!

**Comment by domoritz** [comment]
This will come with https://github.com/vega/ipyvega/issues/77
